doc/api/n-api.md

```diff
@@ -0,0 +1,3021 @@
+# N-API
+
+> Stability: 1 - Experimental
+
+
+N-API is an API for building native Addons. It is independent from
```

Comment:
Suggestion: it might be beneficial to have a quick preferred pronunciation guide for `N-API` included if we really don't want people calling it "nappy"...`